conftest.py

```diff
@@ -9,32 +9,41 @@
 from mantidimaging.core.utility.leak_tracker import leak_tracker
 
 
-def _test_gui_system_filename_match(basename: str) -> bool:
-    return "gui_system" in basename and "_test.py" in basename
-
-
 def pytest_addoption(parser):
     parser.addoption("--run-system-tests", action="store_true", default=False, help="Run GUI system tests")
+    parser.addoption("--run-unit-tests", action="store_true", default=False, help="Run unit tests")
+    parser.addoption("--run-eyes-tests", action="store_true", default=False, help="Run eyes tests")
 
 
 def pytest_configure(config):
     config.addinivalue_line("markers", "system: GUI system tests")
+    config.addinivalue_line("markers", "unit: unit tests")
+    config.addinivalue_line("markers", "eyes: eyes tests")
 
 
-def pytest_ignore_collect(path, config):
-    # When running GUI system tests, ignore all other files
-    if config.getoption("--run-system-tests") and path.isfile() and not _test_gui_system_filename_match(path.basename):
-        return True
-    else:
-        return False
+allowed_markers = []
+skipped_tests = []
 
 
 def pytest_collection_modifyitems(config, items):
-    if not config.getoption("--run-system-tests"):
-        skip_system = pytest.mark.skip(reason="use --run-system-tests option to run")
-        for item in items:
-            if "system" in item.keywords:
-                item.add_marker(skip_system)
+    if config.getoption("--run-system-tests"):
+        allowed_markers.append(pytest.mark.system.mark)
+    if config.getoption("--run-unit-tests"):
+        allowed_markers.append(pytest.mark.unit.mark)
+    if config.getoption("--run-eyes-tests"):
+        allowed_markers.append(pytest.mark.eyes.mark)
```

Comment:
Could this be tweaked so that if neither of system or eyes is selected, then unit tests gets automatically switched on?

e.g.
```python
    if config.getoption("--run-system-tests"):
        allowed_markers.append(pytest.mark.system.mark)
    if config.getoption("--run-eyes-tests"):
        allowed_markers.append(pytest.mark.eyes.mark)
    if config.getoption("--run-unit-tests") or len(allowed_markers) == 0:
        allowed_markers.append(pytest.mark.unit.mark)
```
It would mean that existing commands would still work.

Reply:
Ah true that's a good way to do it, I'll change that